e2e/scripts/run-e2e.sh

```diff
@@ -55,32 +49,60 @@ function _get_test(){
     fi
 }
 
+# run_test runs a single E2E test.
+function run_test() {
+  # if the dev configs directory is present, enable fzf completion to select a test config file to use.
+
+  # if test is set, that is used directly, otherwise the test can be interactively provided if fzf is installed.
+  TEST="$(_get_test ${TEST})"
+
+  # if jq is installed, we can automatically determine the test entrypoint.
+  if command -v jq > /dev/null; then
+     cd ..
+     ENTRY_POINT="$(go run -mod=readonly cmd/build_test_matrix/main.go | jq -r --arg TEST "${TEST}" '.include[] | select( .test == $TEST)  | .entrypoint')"
+     cd - > /dev/null
+  fi
+
+
+  # find the name of the file that has this test in it.
+  test_file="$(grep --recursive --files-with-matches './' -e "${TEST}()")"
+
+  # we run the test on the directory as specific files may reference types in other files but within the package.
+  test_dir="$(dirname $test_file)"
+
+  # run the test file directly, this allows log output to be streamed directly in the terminal sessions
+  # without needed to wait for the test to finish.
+  # it shouldn't take 30m, but the wasm test can be quite slow, so we can be generous.
+  go test -v "${test_dir}" --run ${ENTRY_POINT} -testify.m ^${TEST}$ -timeout 30m
+}
+
+# run_suite runs a full E2E test suite.
+function run_suite() {
+  # if jq is installed, we can automatically determine the test entrypoint.
+  if command -v jq > /dev/null; then
+     cd ..
+     ENTRY_POINT="$(go run -mod=readonly cmd/build_test_matrix/main.go | jq  -r '.include[] | .entrypoint' | uniq | fzf)"
+     cd - > /dev/null
+  fi
+
+  # find the name of the file that has this test in it.
+  test_file="$(grep --recursive --files-with-matches './tests' -e "${ENTRY_POINT}")"
+  test_dir="$(dirname $test_file)"
+
+  # TODO: add the -p flag to run tests in parallel
```

Comment:
[q] do we have an issue to make sure we don't forget? :)

Reply:
all this stuff is kind of building up to close this issue https://github.com/cosmos/ibc-go/pull/5317 , so that won't be complete until we can actually run tests in parallel